fem/bilininteg.cpp

```diff
@@ -3654,6 +3652,13 @@ void DGDiffusionIntegrator::AssembleFaceMatrix(
    }
 }
 
+const IntegrationRule &DGDiffusionIntegrator::GetRule(
+   int order, FaceElementTransformations &T)
+{
+   // order is typically the maximum of the order of the left and right elements
+   // neighboring the given face.
+   return irs.Get(T.GetGeometryType(), 2*order);
```

Comment:
The variable `irs` uses `Quadrature1D::GaussLobatto` -- this changes the current behavior for non-PA and I don't think that is necessary.

For PA, the use of `GaussLobatto` quadrature seems intentional and maybe necessary. If that's the case, I'd suggest defining separate methods: `GetRule` and `GetRulePA` -- this first used for non-PA, the second for PA.

Reply:
Yes, the use of Gauss-Lobatto quadrature for PA is currently necessary, because we need to get the element Jacobians at the face quadrature points. The way we do that is to get the mesh geometric factors at the Gauss-Lobatto points, and then extract the quadrature points that lie on each face. It is possible to extend this to work for arbitrary quadrature rules either by interpolating, or by extending the functionality of the QuadratureInterpolator, but it's not completely straightforward.

I changed the code in 1c0bcaf7c so that non-PA uses the default `IntRules`, and PA uses Gauss-Lobatto of equivalent order. This has the side-effect that the PA and non-PA versions may give different answers if the quadrature is not exact.

Reply:
Looks good. Thanks!